.circleci/jobs/build_and_deploy.yml

```
@@ -107,11 +107,103 @@ jobs:
             - run:
                   name: Remove MySQL port translation (see BH-664)
                   command: yq delete --inplace docker-compose.yml services.mysql.ports
+            -   run:
+                    name: Setup tests results folder and log folder
+                    command: mkdir -p var/tests/phpspec var/tests/csfixer var/logs var/tests/screenshots ~/.composer
+            -   run:
+                    name: Change owner on project dir (docker needs uid 1000, circleci can be another uid)
+                    command: sudo chown -R 1000:1000 ../project
+            -   run:
+                    name: Change permissions on project dir
+                    command: sudo chmod -R 777 ../project
             - persist_to_workspace:
                   root: ~/
                   paths:
                       - project
 
+    install_back_dependencies:
+        machine:
+            image: *executor-machine
+        steps:
+            -   attach_workspace:
+                    at: ~/
+            -   run:
+                    name: Creating cache key for back dependencies
+                    command: cat composer.json > ~/back-dependency.hash && date +%F >> ~/back-dependency.hash
+            -   restore_cache:
+                    name: Restore cache - back dependencies
+                    keys:
+                        - backend-dependency-cache-{{ .Environment.CACHE_VERSION }}-{{ checksum "~/back-dependency.hash" }}
+            -   run:
+                    name: Pull docker image for php
+                    command: docker-compose pull php
+            -   run:
+                    name: Change owner on project dir (docker needs uid 1000, circleci can be another uid)
+                    command: sudo chown -R 1000:1000 ../project
+            -   run:
+                    name: Change permissions on project dir
+                    command: sudo chmod -R 777 ../project
+            -   run:
+                    name: Install back dependencies
+                    command: make vendor
```

- run:
```suggestion
            - run:
                  name: Copy docker-compose.override.yml.dist
command: cp .circleci/docker-compose.override.yml.dist docker-compose.override.yml
```
            for inmemory database